src/components/table/utils.ts

```diff
@@ -30,3 +38,113 @@ export const getColumnsKeys = (columns: Column[]): string[] => {
 export const toggleDirection = (direction: SortingDirection) => {
   return isAsc(direction) ? DESC : ASC;
 };
+
+export const isPrimaryColumn = (column: PrimaryColumn | FixedColumn): boolean => {
+  return 'primary' in column && column.primary === true;
+};
+
+export const getRowSizeClassName = (rowData: RowData): string => {
+  const rowSize = rowData.rowConfigs?.size ?? 'default';
+  return `size-${rowSize}`;
+};
+
+export const calculatePinnedPosition = (
+  columnIndex: number,
+  columns: (PrimaryColumn | FixedColumn)[],
+  columnWidthsRef: React.MutableRefObject<Map<string, number>>,
+  isRowsExpandable: boolean,
+  selectable: boolean,
+): number => {
+  let position = 0;
+
+  if (isRowsExpandable) {
+    position += EXPANDABLE_CHECKBOX_COLUMN_WIDTH;
+  }
+
+  if (selectable) {
+    position += EXPANDABLE_CHECKBOX_COLUMN_WIDTH;
+  }
+
+  for (let i = 0; i < columnIndex; i++) {
+    const column = columns[i];
+    if (isPrimaryColumn(column)) {
+      const actualWidth = columnWidthsRef.current.get(column.key) || PRIMARY_COLUMN_DEFAULT_WIDTH;
+      position += actualWidth;
+    } else {
+      const fixedColumn = column as FixedColumn;
+      const width = isString(fixedColumn.width)
+        ? parseInt(fixedColumn.width, 10)
+        : fixedColumn.width;
+      position += width;
+    }
+  }
+  return position;
+};
+
+export const getCellStyle = (
+  column: FixedColumn | PrimaryColumn,
+  isPinned: boolean,
+  pinnedIndex: number | undefined,
+  pinnedColumns: (PrimaryColumn | FixedColumn)[],
+  columnWidthsRef: React.MutableRefObject<Map<string, number>>,
+  isRowsExpandable: boolean,
+  selectable: boolean,
+): CSSProperties => {
+  const baseStyle: CSSProperties = {};
+
+  if (!isPrimaryColumn(column)) {
+    const fixedColumn = column as FixedColumn;
+    baseStyle.textAlign = fixedColumn.align;
+  }
+
+  if (isPinned && pinnedIndex !== undefined) {
+    const leftPosition = calculatePinnedPosition(
+      pinnedIndex,
+      pinnedColumns,
+      columnWidthsRef,
+      isRowsExpandable,
+      selectable,
+    );
+    baseStyle.left = `${leftPosition}px`;
+  }
+
+  return baseStyle;
+};
+
+export const getGridTemplateColumns = (
+  pinnedColumns: (PrimaryColumn | FixedColumn)[],
+  scrollableColumns: (PrimaryColumn | FixedColumn)[],
+  isRowsExpandable: boolean,
+  selectable: boolean,
+  renderRowActions: boolean,
+  isHeader = false,
+): string => {
+  const columns: string[] = [];
+
+  if (isRowsExpandable) {
+    columns.push(`${EXPANDABLE_CHECKBOX_COLUMN_WIDTH}px`);
+  }
+
+  if (isHeader && selectable) {
+    columns.push(`${EXPANDABLE_CHECKBOX_COLUMN_WIDTH}px`);
+  }
+
+  const addColumnWidth = (column: Column) => {
+    if (isPrimaryColumn(column as PrimaryColumn | FixedColumn)) {
```

Comment:
We can get rid of type assertion here as we use the function for `pinnedColumns` and `scrollableColumns` only, that have type `PrimaryColumn | FixedColumn`

```suggestion
  const addColumnWidth = (column: PrimaryColumn | FixedColumn) => {
    if (isPrimaryColumn(column)) {
```